src/node_port.py

```diff
@@ -4,13 +4,14 @@
 
 """Manager for handling mongos Kubernetes resources for a single mongos pod."""
 
+from typing import Optional
```

Comment:
if you have [reviewed node port for external clients](https://github.com/canonical/mongos-k8s-operator/pull/32) then these changes are the same